src/apps/chifra/pkg/validate/blockvalidate_test.go

```diff
@@ -29,26 +30,79 @@ func TestIsBlockHash(t *testing.T) {
 }
 
 func TestIsBlockNumber(t *testing.T) {
-	if IsBlockNumber("notanumber") {
+	if check, _ := IsBlockNumber("notanumber"); check {
 		t.Error("Passes for strings")
 	}
 
-	if IsBlockNumber("") {
+	if check, _ := IsBlockNumber(""); check {
 		t.Error("Passes for empty string")
 	}
 
 	// We only support decimal numbers for block numbers
-	if !IsBlockNumber("0xff") {
+	if check, _ := IsBlockNumber("0xff"); !check {
 		t.Error("Fails for hex")
 	}
 
-	if !IsBlockNumber("0") {
+	if check, _ := IsBlockNumber("0"); !check {
 		t.Error("Fails for 0")
 	}
 
-	if !IsBlockNumber("1000") {
+	if check, _ := IsBlockNumber("1000"); !check {
 		t.Error("Fails for numbers")
 	}
+
+	check, value := IsBlockNumber("12147043")
+	if !check || value != uint32(12147043) {
+		t.Fatal("Failed for correct input:", check, value)
+	}
+}
+
+func TestIsBlockNumberList(t *testing.T) {
+	type args struct {
+		strs []string
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  bool
+		want1 []blockNum
+	}{
+		{
+			name: "list of valid blocknumbers",
+			args: args{
+				strs: []string{"12147043", "12147042", "12147041"},
+			},
+			want:  true,
+			want1: []blockNum{12147043, 12147042, 12147041},
+		},
+		{
+			name: "list of invalid blocknumbers",
+			args: args{
+				strs: []string{"0xfff", "latest", "newest"},
```

Comment:
0xfff and latest should both return valid block numbers

Reply:
I'm not sure why we don't support hex block numbers (I remember the discussion about it and there's a comment in the code that we don't want to support them, but I'm not sure why), but when it comes to `latest` I think it should return a block number when it's parsed, while this function should return false (because `latest` is a special name)